src/Compilers/CSharp/Portable/Symbols/Retargeting/RetargetingSymbolTranslator.cs

```diff
@@ -826,68 +826,54 @@ public static ErrorTypeSymbol Retarget(ErrorTypeSymbol type)
 
             public ImmutableArray<Symbol> Retarget(ImmutableArray<Symbol> arr)
             {
-                var symbols = ArrayBuilder<Symbol>.GetInstance(arr.Length);
-
-                foreach (var s in arr)
-                {
-                    symbols.Add(Retarget(s));
-                }
-
-                return symbols.ToImmutableAndFree();
+                return ImmutableArray.CreateRange(
+                    arr,
+                    static (s, self) => self.Retarget(s),
+                    this);
```

Comment:
could you use SelectAsArray instead?  that seems to be the general entrypoint we're already using much more in both compiler and IDE layers.   It's just less cognitive overhead that way.

Reply:
`SelectAsArray` is certainly beneficial for source enumerables that are not already instances of `ImmutableArray<TSource>`, but `ImmutableArray.CreateRange` is the BCL method for this function. `SelectAsArray` has an convenience overload that passes through to it for `ImmutableArray<TSource>`, but I'm not sure that is enough to warrant its use here.